<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 7 Cancer Dependency Map - DepMap | PGT Bioinformatics Course</title>
  <meta name="description" content="Basic introduction to RNAseq analysis using R/Bioconductor" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 7 Cancer Dependency Map - DepMap | PGT Bioinformatics Course" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Basic introduction to RNAseq analysis using R/Bioconductor" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 7 Cancer Dependency Map - DepMap | PGT Bioinformatics Course" />
  
  <meta name="twitter:description" content="Basic introduction to RNAseq analysis using R/Bioconductor" />
  

<meta name="author" content="Peter J. Bailey" />


<meta name="date" content="2021-07-19" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="rnaseq-analysis-deseq2.html"/>

<script src="libs/header-attrs-2.9/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">PGT Bioinformatics Course</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#installing-r-and-rstudio"><i class="fa fa-check"></i><b>1.1</b> Installing R and RStudio</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#installing-r-and-bioconductor-packages"><i class="fa fa-check"></i><b>1.2</b> Installing R and Bioconductor packages</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#loading-r-and-bioconductor-packages"><i class="fa fa-check"></i><b>1.3</b> Loading R and Bioconductor packages</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#learning_resources"><i class="fa fa-check"></i><b>1.4</b> Learning resources</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> Introduction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="intro.html"><a href="intro.html#course-outline"><i class="fa fa-check"></i><b>2.1</b> Course outline</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="intro.html"><a href="intro.html#part1"><i class="fa fa-check"></i><b>2.1.1</b> PART 1</a></li>
<li class="chapter" data-level="2.1.2" data-path="intro.html"><a href="intro.html#part2"><i class="fa fa-check"></i><b>2.1.2</b> PART 2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="r-programming-language-primer.html"><a href="r-programming-language-primer.html"><i class="fa fa-check"></i><b>3</b> R Programming language primer</a>
<ul>
<li class="chapter" data-level="3.1" data-path="r-programming-language-primer.html"><a href="r-programming-language-primer.html#vectors"><i class="fa fa-check"></i><b>3.1</b> Vectors</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="r-programming-language-primer.html"><a href="r-programming-language-primer.html#vectors-in-action"><i class="fa fa-check"></i><b>3.1.1</b> Vectors in action</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="r-programming-language-primer.html"><a href="r-programming-language-primer.html#matrix"><i class="fa fa-check"></i><b>3.2</b> Matrix</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="r-programming-language-primer.html"><a href="r-programming-language-primer.html#matrices-in-action"><i class="fa fa-check"></i><b>3.2.1</b> Matrices in action</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="r-programming-language-primer.html"><a href="r-programming-language-primer.html#data-frame"><i class="fa fa-check"></i><b>3.3</b> Data Frame</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="r-programming-language-primer.html"><a href="r-programming-language-primer.html#data-frames-in-action"><i class="fa fa-check"></i><b>3.3.1</b> Data Frames in action</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="r-programming-language-primer.html"><a href="r-programming-language-primer.html#factors"><i class="fa fa-check"></i><b>3.4</b> Factors</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="r-programming-language-primer.html"><a href="r-programming-language-primer.html#factors-in-action"><i class="fa fa-check"></i><b>3.4.1</b> Factors in action</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="r-programming-language-primer.html"><a href="r-programming-language-primer.html#list"><i class="fa fa-check"></i><b>3.5</b> List</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="r-programming-language-primer.html"><a href="r-programming-language-primer.html#lists-in-action"><i class="fa fa-check"></i><b>3.5.1</b> Lists in action</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="r-programming-language-primer.html"><a href="r-programming-language-primer.html#array"><i class="fa fa-check"></i><b>3.6</b> Array</a></li>
<li class="chapter" data-level="3.7" data-path="r-programming-language-primer.html"><a href="r-programming-language-primer.html#creating-functions"><i class="fa fa-check"></i><b>3.7</b> Creating functions</a></li>
<li class="chapter" data-level="3.8" data-path="r-programming-language-primer.html"><a href="r-programming-language-primer.html#loading-data"><i class="fa fa-check"></i><b>3.8</b> Loading Data</a></li>
<li class="chapter" data-level="3.9" data-path="r-programming-language-primer.html"><a href="r-programming-language-primer.html#saving-data"><i class="fa fa-check"></i><b>3.9</b> Saving Data</a></li>
<li class="chapter" data-level="3.10" data-path="r-programming-language-primer.html"><a href="r-programming-language-primer.html#saving-plots"><i class="fa fa-check"></i><b>3.10</b> Saving plots</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="descriptive-statistics-and-plotting.html"><a href="descriptive-statistics-and-plotting.html"><i class="fa fa-check"></i><b>4</b> Descriptive statistics and plotting</a>
<ul>
<li class="chapter" data-level="4.1" data-path="descriptive-statistics-and-plotting.html"><a href="descriptive-statistics-and-plotting.html#the-cancer-gene-atlas-tcga---pancreatic-ductal-adenocarcinoma-data-set"><i class="fa fa-check"></i><b>4.1</b> The Cancer Gene Atlas (TCGA) - Pancreatic Ductal Adenocarcinoma Data Set</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="descriptive-statistics-and-plotting.html"><a href="descriptive-statistics-and-plotting.html#gene-expression-data"><i class="fa fa-check"></i><b>4.1.1</b> Gene expression data</a></li>
<li class="chapter" data-level="4.1.2" data-path="descriptive-statistics-and-plotting.html"><a href="descriptive-statistics-and-plotting.html#clinical-data"><i class="fa fa-check"></i><b>4.1.2</b> Clinical data</a></li>
<li class="chapter" data-level="4.1.3" data-path="descriptive-statistics-and-plotting.html"><a href="descriptive-statistics-and-plotting.html#pdac-subtypes"><i class="fa fa-check"></i><b>4.1.3</b> PDAC Subtypes</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="descriptive-statistics-and-plotting.html"><a href="descriptive-statistics-and-plotting.html#descriptive-plots"><i class="fa fa-check"></i><b>4.2</b> Descriptive plots</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="descriptive-statistics-and-plotting.html"><a href="descriptive-statistics-and-plotting.html#boxplots"><i class="fa fa-check"></i><b>4.2.1</b> Boxplots</a></li>
<li class="chapter" data-level="4.2.2" data-path="descriptive-statistics-and-plotting.html"><a href="descriptive-statistics-and-plotting.html#scatter-plots"><i class="fa fa-check"></i><b>4.2.2</b> Scatter plots</a></li>
<li class="chapter" data-level="4.2.3" data-path="descriptive-statistics-and-plotting.html"><a href="descriptive-statistics-and-plotting.html#hierarchical-clustering"><i class="fa fa-check"></i><b>4.2.3</b> Hierarchical Clustering</a></li>
<li class="chapter" data-level="4.2.4" data-path="descriptive-statistics-and-plotting.html"><a href="descriptive-statistics-and-plotting.html#hierarchical-clustering-using-the-moffitt-transcriptomic-signature"><i class="fa fa-check"></i><b>4.2.4</b> Hierarchical clustering using the Moffitt transcriptomic signature</a></li>
<li class="chapter" data-level="4.2.5" data-path="descriptive-statistics-and-plotting.html"><a href="descriptive-statistics-and-plotting.html#heatmaps"><i class="fa fa-check"></i><b>4.2.5</b> Heatmaps</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="rnaseq-analysis.html"><a href="rnaseq-analysis.html"><i class="fa fa-check"></i><b>5</b> RNAseq Analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="rnaseq-analysis.html"><a href="rnaseq-analysis.html#rnaseq-downstream-analysis"><i class="fa fa-check"></i><b>5.1</b> RNAseq downstream analysis</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="rnaseq-analysis.html"><a href="rnaseq-analysis.html#nf-corernaseq-pipeline---installation"><i class="fa fa-check"></i><b>5.1.1</b> nf-core/rnaseq pipeline - installation</a></li>
<li class="chapter" data-level="5.1.2" data-path="rnaseq-analysis.html"><a href="rnaseq-analysis.html#nextflow"><i class="fa fa-check"></i><b>5.1.2</b> Nextflow</a></li>
<li class="chapter" data-level="5.1.3" data-path="rnaseq-analysis.html"><a href="rnaseq-analysis.html#additional-pipeline-software"><i class="fa fa-check"></i><b>5.1.3</b> Additional pipeline software</a></li>
<li class="chapter" data-level="5.1.4" data-path="rnaseq-analysis.html"><a href="rnaseq-analysis.html#pipeline-code"><i class="fa fa-check"></i><b>5.1.4</b> Pipeline code</a></li>
<li class="chapter" data-level="5.1.5" data-path="rnaseq-analysis.html"><a href="rnaseq-analysis.html#reference-genomes"><i class="fa fa-check"></i><b>5.1.5</b> Reference genomes</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="rnaseq-analysis.html"><a href="rnaseq-analysis.html#running-the-nf-corernaseq-pipeline"><i class="fa fa-check"></i><b>5.2</b> Running the nf-core/rnaseq pipeline</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="rnaseq-analysis.html"><a href="rnaseq-analysis.html#setup"><i class="fa fa-check"></i><b>5.2.1</b> Setup</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="rnaseq-analysis.html"><a href="rnaseq-analysis.html#running-the-nf-corernaseq-pipeline-1"><i class="fa fa-check"></i><b>5.3</b> Running the nf-core/rnaseq pipeline</a></li>
<li class="chapter" data-level="5.4" data-path="rnaseq-analysis.html"><a href="rnaseq-analysis.html#quality-control-qc-and-merged-count-files"><i class="fa fa-check"></i><b>5.4</b> Quality Control (QC) and Merged count files</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="rnaseq-analysis.html"><a href="rnaseq-analysis.html#qc-metrics"><i class="fa fa-check"></i><b>5.4.1</b> QC Metrics</a></li>
<li class="chapter" data-level="5.4.2" data-path="rnaseq-analysis.html"><a href="rnaseq-analysis.html#merged-count-files"><i class="fa fa-check"></i><b>5.4.2</b> Merged count files</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="rnaseq-analysis-deseq2.html"><a href="rnaseq-analysis-deseq2.html"><i class="fa fa-check"></i><b>6</b> RNAseq Analysis - DESeq2</a>
<ul>
<li class="chapter" data-level="6.1" data-path="rnaseq-analysis-deseq2.html"><a href="rnaseq-analysis-deseq2.html#deseq2-package"><i class="fa fa-check"></i><b>6.1</b> DESEq2 package</a></li>
<li class="chapter" data-level="6.2" data-path="rnaseq-analysis-deseq2.html"><a href="rnaseq-analysis-deseq2.html#the-deseqdataset"><i class="fa fa-check"></i><b>6.2</b> The DESeqDataSet</a></li>
<li class="chapter" data-level="6.3" data-path="rnaseq-analysis-deseq2.html"><a href="rnaseq-analysis-deseq2.html#count-matrix-input"><i class="fa fa-check"></i><b>6.3</b> Count matrix input</a></li>
<li class="chapter" data-level="6.4" data-path="rnaseq-analysis-deseq2.html"><a href="rnaseq-analysis-deseq2.html#pre-filtering"><i class="fa fa-check"></i><b>6.4</b> Pre-filtering</a></li>
<li class="chapter" data-level="6.5" data-path="rnaseq-analysis-deseq2.html"><a href="rnaseq-analysis-deseq2.html#differential-expression-analysis"><i class="fa fa-check"></i><b>6.5</b> Differential expression analysis</a></li>
<li class="chapter" data-level="6.6" data-path="rnaseq-analysis-deseq2.html"><a href="rnaseq-analysis-deseq2.html#gene-annotation"><i class="fa fa-check"></i><b>6.6</b> Gene Annotation</a></li>
<li class="chapter" data-level="6.7" data-path="rnaseq-analysis-deseq2.html"><a href="rnaseq-analysis-deseq2.html#exploring-the-results"><i class="fa fa-check"></i><b>6.7</b> Exploring the results</a></li>
<li class="chapter" data-level="6.8" data-path="rnaseq-analysis-deseq2.html"><a href="rnaseq-analysis-deseq2.html#exporting-results"><i class="fa fa-check"></i><b>6.8</b> Exporting results</a></li>
<li class="chapter" data-level="6.9" data-path="rnaseq-analysis-deseq2.html"><a href="rnaseq-analysis-deseq2.html#data-transformation-and-visualisation"><i class="fa fa-check"></i><b>6.9</b> Data transformation and visualisation</a>
<ul>
<li class="chapter" data-level="6.9.1" data-path="rnaseq-analysis-deseq2.html"><a href="rnaseq-analysis-deseq2.html#pca-using-transformed-data"><i class="fa fa-check"></i><b>6.9.1</b> PCA using transformed data</a></li>
<li class="chapter" data-level="6.9.2" data-path="rnaseq-analysis-deseq2.html"><a href="rnaseq-analysis-deseq2.html#heatmaps-using-transformed-data"><i class="fa fa-check"></i><b>6.9.2</b> Heatmaps using transformed data</a></li>
</ul></li>
<li class="chapter" data-level="6.10" data-path="rnaseq-analysis-deseq2.html"><a href="rnaseq-analysis-deseq2.html#gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>6.10</b> Gene Set Enrichment Analysis</a>
<ul>
<li class="chapter" data-level="6.10.1" data-path="rnaseq-analysis-deseq2.html"><a href="rnaseq-analysis-deseq2.html#kegg"><i class="fa fa-check"></i><b>6.10.1</b> KEGG</a></li>
<li class="chapter" data-level="6.10.2" data-path="rnaseq-analysis-deseq2.html"><a href="rnaseq-analysis-deseq2.html#reactome"><i class="fa fa-check"></i><b>6.10.2</b> REACTOME</a></li>
<li class="chapter" data-level="6.10.3" data-path="rnaseq-analysis-deseq2.html"><a href="rnaseq-analysis-deseq2.html#plotting-functions"><i class="fa fa-check"></i><b>6.10.3</b> Plotting functions</a></li>
<li class="chapter" data-level="6.10.4" data-path="rnaseq-analysis-deseq2.html"><a href="rnaseq-analysis-deseq2.html#extracting-enrichment-results-to-generate-heatmaps"><i class="fa fa-check"></i><b>6.10.4</b> Extracting enrichment results to generate heatmaps</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="cancer-dependency-map-depmap.html"><a href="cancer-dependency-map-depmap.html"><i class="fa fa-check"></i><b>7</b> Cancer Dependency Map - DepMap</a>
<ul>
<li class="chapter" data-level="7.1" data-path="cancer-dependency-map-depmap.html"><a href="cancer-dependency-map-depmap.html#downloading-depmap-data-using-the-depmap-package"><i class="fa fa-check"></i><b>7.1</b> Downloading DepMap data using the <code>depmap</code> package</a></li>
<li class="chapter" data-level="7.2" data-path="cancer-dependency-map-depmap.html"><a href="cancer-dependency-map-depmap.html#exploring-depmap-transcriptomic-data"><i class="fa fa-check"></i><b>7.2</b> Exploring DepMap transcriptomic data</a></li>
<li class="chapter" data-level="7.3" data-path="cancer-dependency-map-depmap.html"><a href="cancer-dependency-map-depmap.html#exploring-depmap-mutation-calls"><i class="fa fa-check"></i><b>7.3</b> Exploring DepMap mutation calls</a></li>
<li class="chapter" data-level="7.4" data-path="cancer-dependency-map-depmap.html"><a href="cancer-dependency-map-depmap.html#exploring-depmap-crispr-data"><i class="fa fa-check"></i><b>7.4</b> Exploring DepMap CRISPR data</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">PGT Bioinformatics Course</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="cancer-dependency-map---depmap" class="section level1" number="7">
<h1><span class="header-section-number">Chapter 7</span> Cancer Dependency Map - DepMap</h1>
<p><a href="https://depmap.org/portal/"><img src="images/depmap_site.png" alt="Cover image" /></a></p>
<p><strong>Chapter 7</strong> will explore the Cancer Dependency Map or DepMap data resource. DepMap provides a comprehensive set of data including genomic information and screening data for gene and small molecule perturbations across hundreds of cancer cell line models. By triangulating information from these and other large-scale datasets, the hope is to define a landscape of genetic targets for therapeutic development, identify patients who respond to these therapies, and develop a better understanding of the vulnerabilities of cancer.</p>
<div id="downloading-depmap-data-using-the-depmap-package" class="section level2" number="7.1">
<h2><span class="header-section-number">7.1</span> Downloading DepMap data using the <code>depmap</code> package</h2>
<p>DepMap data can downloaded directly from the DepMap data portal website. Alternatively, DepMap data can be downloaded from within R using the Bioconductor package <code>depmap</code>.</p>
<p><code>depmap</code> provides easy access to datasets from the most current DepMap release, including RNAi and CRISPR-Cas9 gene knockout screens quantifying the genetic dependency for select cancer cell lines.</p>
<p>Additional datasets are also available including log copy number of genes for select cell lines, protein expression of cell lines as measured by reverse phase protein lysate microarray (RPPA), ‘Transcript Per Million’ (TPM) data, as well as supplementary datasets which contain metadata and mutation calls for the other datasets found in the current release.</p>
<p>The 19Q3 release adds the drug_dependency dataset, that contains cancer cell line dependency data with respect to drug and drug-candidate compounds.</p>
<p>The 20Q2 release adds the proteomic dataset that contains quantitative profiling of proteins via mass spectrometry.</p>
<p>First install the <code>depmap</code> package as follows:</p>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="cancer-dependency-map-depmap.html#cb87-1" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;depmap&quot;</span>)</span></code></pre></div>
<p>Load the <code>depmap</code> and <code>ExperimentHub</code> libraries into your working environment:</p>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="cancer-dependency-map-depmap.html#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;depmap&quot;</span>)</span>
<span id="cb88-2"><a href="cancer-dependency-map-depmap.html#cb88-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ExperimentHub&quot;</span>)</span></code></pre></div>
<p>First we create an ExperimentHub object called <code>eh</code> and generate a query using the <code>depmap</code> argument to obtain a list of DepMap data that are available for download:</p>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="cancer-dependency-map-depmap.html#cb89-1" aria-hidden="true" tabindex="-1"></a>eh <span class="ot">&lt;-</span> <span class="fu">ExperimentHub</span>()</span>
<span id="cb89-2"><a href="cancer-dependency-map-depmap.html#cb89-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; snapshotDate(): 2021-05-18</span></span>
<span id="cb89-3"><a href="cancer-dependency-map-depmap.html#cb89-3" aria-hidden="true" tabindex="-1"></a><span class="fu">query</span>(eh, <span class="st">&quot;depmap&quot;</span>)</span>
<span id="cb89-4"><a href="cancer-dependency-map-depmap.html#cb89-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ExperimentHub with 58 records</span></span>
<span id="cb89-5"><a href="cancer-dependency-map-depmap.html#cb89-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # snapshotDate(): 2021-05-18</span></span>
<span id="cb89-6"><a href="cancer-dependency-map-depmap.html#cb89-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # $dataprovider: Broad Institute</span></span>
<span id="cb89-7"><a href="cancer-dependency-map-depmap.html#cb89-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # $species: Homo sapiens</span></span>
<span id="cb89-8"><a href="cancer-dependency-map-depmap.html#cb89-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # $rdataclass: tibble</span></span>
<span id="cb89-9"><a href="cancer-dependency-map-depmap.html#cb89-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # additional mcols(): taxonomyid, genome, description,</span></span>
<span id="cb89-10"><a href="cancer-dependency-map-depmap.html#cb89-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass,</span></span>
<span id="cb89-11"><a href="cancer-dependency-map-depmap.html#cb89-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   tags, rdatapath, sourceurl, sourcetype </span></span>
<span id="cb89-12"><a href="cancer-dependency-map-depmap.html#cb89-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # retrieve records with, e.g., &#39;object[[&quot;EH2260&quot;]]&#39; </span></span>
<span id="cb89-13"><a href="cancer-dependency-map-depmap.html#cb89-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb89-14"><a href="cancer-dependency-map-depmap.html#cb89-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;            title             </span></span>
<span id="cb89-15"><a href="cancer-dependency-map-depmap.html#cb89-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   EH2260 | rnai_19Q1         </span></span>
<span id="cb89-16"><a href="cancer-dependency-map-depmap.html#cb89-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   EH2261 | crispr_19Q1       </span></span>
<span id="cb89-17"><a href="cancer-dependency-map-depmap.html#cb89-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   EH2262 | copyNumber_19Q1   </span></span>
<span id="cb89-18"><a href="cancer-dependency-map-depmap.html#cb89-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   EH2263 | RPPA_19Q1         </span></span>
<span id="cb89-19"><a href="cancer-dependency-map-depmap.html#cb89-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   EH2264 | TPM_19Q1          </span></span>
<span id="cb89-20"><a href="cancer-dependency-map-depmap.html#cb89-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ...      ...               </span></span>
<span id="cb89-21"><a href="cancer-dependency-map-depmap.html#cb89-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   EH6118 | crispr_21Q2       </span></span>
<span id="cb89-22"><a href="cancer-dependency-map-depmap.html#cb89-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   EH6119 | copyNumber_21Q2   </span></span>
<span id="cb89-23"><a href="cancer-dependency-map-depmap.html#cb89-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   EH6120 | TPM_21Q2          </span></span>
<span id="cb89-24"><a href="cancer-dependency-map-depmap.html#cb89-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   EH6121 | mutationCalls_21Q2</span></span>
<span id="cb89-25"><a href="cancer-dependency-map-depmap.html#cb89-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   EH6122 | metadata_21Q2</span></span></code></pre></div>
<p>To retrieve a dataset of interest we can simply use the name of the list element. For example, let’s obtain the DepMap transcriptomic data as follows:</p>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="cancer-dependency-map-depmap.html#cb90-1" aria-hidden="true" tabindex="-1"></a>TPM <span class="ot">&lt;-</span> eh[[<span class="st">&quot;EH6120&quot;</span>]]</span>
<span id="cb90-2"><a href="cancer-dependency-map-depmap.html#cb90-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; see ?depmap and browseVignettes(&#39;depmap&#39;) for documentation</span></span>
<span id="cb90-3"><a href="cancer-dependency-map-depmap.html#cb90-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; downloading 1 resources</span></span>
<span id="cb90-4"><a href="cancer-dependency-map-depmap.html#cb90-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; retrieving 1 resource</span></span>
<span id="cb90-5"><a href="cancer-dependency-map-depmap.html#cb90-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; loading from cache</span></span>
<span id="cb90-6"><a href="cancer-dependency-map-depmap.html#cb90-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(TPM)</span>
<span id="cb90-7"><a href="cancer-dependency-map-depmap.html#cb90-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 6 x 6</span></span>
<span id="cb90-8"><a href="cancer-dependency-map-depmap.html#cb90-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   depmap_id  gene        rna_expression entrez_id gene_name cell_line        </span></span>
<span id="cb90-9"><a href="cancer-dependency-map-depmap.html#cb90-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;      &lt;chr&gt;                &lt;dbl&gt;     &lt;int&gt; &lt;chr&gt;     &lt;chr&gt;            </span></span>
<span id="cb90-10"><a href="cancer-dependency-map-depmap.html#cb90-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 ACH-001113 TSPAN6 (71…           4.99      7105 TSPAN6    LC1SQSF_LUNG     </span></span>
<span id="cb90-11"><a href="cancer-dependency-map-depmap.html#cb90-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 ACH-001289 TSPAN6 (71…           5.21      7105 TSPAN6    COGAR359_SOFT_TI…</span></span>
<span id="cb90-12"><a href="cancer-dependency-map-depmap.html#cb90-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3 ACH-001339 TSPAN6 (71…           3.78      7105 TSPAN6    COLO794_SKIN     </span></span>
<span id="cb90-13"><a href="cancer-dependency-map-depmap.html#cb90-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4 ACH-001538 TSPAN6 (71…           5.73      7105 TSPAN6    KKU213_BILIARY_T…</span></span>
<span id="cb90-14"><a href="cancer-dependency-map-depmap.html#cb90-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5 ACH-000242 TSPAN6 (71…           7.47      7105 TSPAN6    RT4_URINARY_TRACT</span></span>
<span id="cb90-15"><a href="cancer-dependency-map-depmap.html#cb90-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6 ACH-000708 TSPAN6 (71…           4.91      7105 TSPAN6    SNU283_LARGE_INT…</span></span></code></pre></div>
<p>We can also obtain detailed information about each of the cell lines by downloading the metadata:</p>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="cancer-dependency-map-depmap.html#cb91-1" aria-hidden="true" tabindex="-1"></a>metaData <span class="ot">&lt;-</span> eh[[<span class="st">&quot;EH6122&quot;</span>]]</span>
<span id="cb91-2"><a href="cancer-dependency-map-depmap.html#cb91-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; see ?depmap and browseVignettes(&#39;depmap&#39;) for documentation</span></span>
<span id="cb91-3"><a href="cancer-dependency-map-depmap.html#cb91-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; loading from cache</span></span>
<span id="cb91-4"><a href="cancer-dependency-map-depmap.html#cb91-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(metaData)</span>
<span id="cb91-5"><a href="cancer-dependency-map-depmap.html#cb91-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 6 x 26</span></span>
<span id="cb91-6"><a href="cancer-dependency-map-depmap.html#cb91-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   depmap_id cell_line_name stripped_cell_l… cell_line aliases cosmic_id sex  </span></span>
<span id="cb91-7"><a href="cancer-dependency-map-depmap.html#cb91-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;     &lt;chr&gt;          &lt;chr&gt;            &lt;chr&gt;     &lt;chr&gt;       &lt;dbl&gt; &lt;chr&gt;</span></span>
<span id="cb91-8"><a href="cancer-dependency-map-depmap.html#cb91-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 ACH-0000… NIH:OVCAR-3    NIHOVCAR3        NIHOVCAR… OVCAR3     905933 Fema…</span></span>
<span id="cb91-9"><a href="cancer-dependency-map-depmap.html#cb91-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 ACH-0000… HL-60          HL60             HL60_HAE… &lt;NA&gt;       905938 Fema…</span></span>
<span id="cb91-10"><a href="cancer-dependency-map-depmap.html#cb91-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3 ACH-0000… CACO2          CACO2            CACO2_LA… CACO2,…        NA Male </span></span>
<span id="cb91-11"><a href="cancer-dependency-map-depmap.html#cb91-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4 ACH-0000… HEL            HEL              HEL_HAEM… &lt;NA&gt;       907053 Male </span></span>
<span id="cb91-12"><a href="cancer-dependency-map-depmap.html#cb91-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5 ACH-0000… HEL 92.1.7     HEL9217          HEL9217_… &lt;NA&gt;           NA Male </span></span>
<span id="cb91-13"><a href="cancer-dependency-map-depmap.html#cb91-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6 ACH-0000… MONO-MAC-6     MONOMAC6         MONOMAC6… &lt;NA&gt;       908148 Male </span></span>
<span id="cb91-14"><a href="cancer-dependency-map-depmap.html#cb91-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # … with 19 more variables: source &lt;chr&gt;, Achilles_n_replicates &lt;dbl&gt;,</span></span>
<span id="cb91-15"><a href="cancer-dependency-map-depmap.html#cb91-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   cell_line_NNMD &lt;dbl&gt;, culture_type &lt;chr&gt;, culture_medium &lt;chr&gt;,</span></span>
<span id="cb91-16"><a href="cancer-dependency-map-depmap.html#cb91-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   cas9_activity &lt;dbl&gt;, RRID &lt;chr&gt;, WTSI_master_cell_ID &lt;dbl&gt;,</span></span>
<span id="cb91-17"><a href="cancer-dependency-map-depmap.html#cb91-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   sample_collection_site &lt;chr&gt;, primary_or_metastasis &lt;chr&gt;,</span></span>
<span id="cb91-18"><a href="cancer-dependency-map-depmap.html#cb91-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   primary_disease &lt;chr&gt;, subtype_disease &lt;chr&gt;, age &lt;dbl&gt;,</span></span>
<span id="cb91-19"><a href="cancer-dependency-map-depmap.html#cb91-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   sanger_id &lt;chr&gt;, additional_info &lt;lgl&gt;, lineage &lt;chr&gt;,</span></span>
<span id="cb91-20"><a href="cancer-dependency-map-depmap.html#cb91-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   lineage_subtype &lt;chr&gt;, lineage_sub_subtype &lt;chr&gt;,</span></span>
<span id="cb91-21"><a href="cancer-dependency-map-depmap.html#cb91-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   lineage_molecular_subtype &lt;chr&gt;</span></span></code></pre></div>
<p>We might for instance be interested in obtaining the lineage of each cell line:</p>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="cancer-dependency-map-depmap.html#cb92-1" aria-hidden="true" tabindex="-1"></a>lineages <span class="ot">&lt;-</span> metaData <span class="sc">%&gt;%</span> <span class="fu">select</span>(lineage) <span class="sc">%&gt;%</span> <span class="fu">unique</span>()</span>
<span id="cb92-2"><a href="cancer-dependency-map-depmap.html#cb92-2" aria-hidden="true" tabindex="-1"></a>lineages<span class="sc">$</span>lineage</span>
<span id="cb92-3"><a href="cancer-dependency-map-depmap.html#cb92-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [1] &quot;ovary&quot;                             &quot;blood&quot;                            </span></span>
<span id="cb92-4"><a href="cancer-dependency-map-depmap.html#cb92-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [3] &quot;colorectal&quot;                        &quot;skin&quot;                             </span></span>
<span id="cb92-5"><a href="cancer-dependency-map-depmap.html#cb92-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [5] &quot;lung&quot;                              &quot;urinary_tract&quot;                    </span></span>
<span id="cb92-6"><a href="cancer-dependency-map-depmap.html#cb92-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  ...</span></span>
<span id="cb92-7"><a href="cancer-dependency-map-depmap.html#cb92-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [35] &quot;adrenal_cortex&quot;                    &quot;epidermoid_carcinoma&quot;             </span></span>
<span id="cb92-8"><a href="cancer-dependency-map-depmap.html#cb92-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [37] &quot;embryo&quot;                            &quot;engineered&quot;                       </span></span>
<span id="cb92-9"><a href="cancer-dependency-map-depmap.html#cb92-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [39] &quot;engineered_blood&quot;</span></span></code></pre></div>
<p>Have you noticed a new way of coding? We are using the <code>dplyr</code> package. What are these <code>%&gt;%</code> functions? <code>%&gt;%</code> functions are pipes that allow us to pass data from one function to another without the hassle of assigning multiple objects.</p>
<p>The <code>dplyr</code> package includes a number of extremely useful functions that allow us to join different tables and integrate related datasets!</p>
</div>
<div id="exploring-depmap-transcriptomic-data" class="section level2" number="7.2">
<h2><span class="header-section-number">7.2</span> Exploring DepMap transcriptomic data</h2>
<p>What if we are interested in examining GATA6 expression across cancer cell lineages? How can we achieve this? Firstly, we notice that the lineage specific information is found in the <code>metaData</code> object while the gene expression data is located in the <code>TPM</code> object. We can join the tables using the <code>dplyr::full_join</code> function as follows:</p>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="cancer-dependency-map-depmap.html#cb93-1" aria-hidden="true" tabindex="-1"></a>expBylineage <span class="ot">&lt;-</span> metaData <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(depmap_id, lineage) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">full_join</span>(TPM, <span class="at">by =</span> <span class="st">&quot;depmap_id&quot;</span>) </span>
<span id="cb93-2"><a href="cancer-dependency-map-depmap.html#cb93-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(expBylineage)</span>
<span id="cb93-3"><a href="cancer-dependency-map-depmap.html#cb93-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 6 x 7</span></span>
<span id="cb93-4"><a href="cancer-dependency-map-depmap.html#cb93-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   depmap_id  lineage gene      rna_expression entrez_id gene_name cell_line  </span></span>
<span id="cb93-5"><a href="cancer-dependency-map-depmap.html#cb93-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;      &lt;chr&gt;   &lt;chr&gt;              &lt;dbl&gt;     &lt;int&gt; &lt;chr&gt;     &lt;chr&gt;      </span></span>
<span id="cb93-6"><a href="cancer-dependency-map-depmap.html#cb93-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 ACH-000001 ovary   TSPAN6 (…         5.82        7105 TSPAN6    NIHOVCAR3_…</span></span>
<span id="cb93-7"><a href="cancer-dependency-map-depmap.html#cb93-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 ACH-000001 ovary   TNMD (64…         0          64102 TNMD      NIHOVCAR3_…</span></span>
<span id="cb93-8"><a href="cancer-dependency-map-depmap.html#cb93-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3 ACH-000001 ovary   DPM1 (88…         7.37        8813 DPM1      NIHOVCAR3_…</span></span>
<span id="cb93-9"><a href="cancer-dependency-map-depmap.html#cb93-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4 ACH-000001 ovary   SCYL3 (5…         2.10       57147 SCYL3     NIHOVCAR3_…</span></span>
<span id="cb93-10"><a href="cancer-dependency-map-depmap.html#cb93-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5 ACH-000001 ovary   C1orf112…         4.22       55732 C1orf112  NIHOVCAR3_…</span></span>
<span id="cb93-11"><a href="cancer-dependency-map-depmap.html#cb93-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6 ACH-000001 ovary   FGR (226…         0.0426      2268 FGR       NIHOVCAR3_…</span></span></code></pre></div>
<p>To extract GATA6 gene expression data we can use the <code>dplyr::filter()</code> function:</p>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="cancer-dependency-map-depmap.html#cb94-1" aria-hidden="true" tabindex="-1"></a>gata6Bylineage <span class="ot">&lt;-</span> expBylineage <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(gene_name <span class="sc">==</span> <span class="st">&quot;GATA6&quot;</span>)</span>
<span id="cb94-2"><a href="cancer-dependency-map-depmap.html#cb94-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(gata6Bylineage)</span>
<span id="cb94-3"><a href="cancer-dependency-map-depmap.html#cb94-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 6 x 7</span></span>
<span id="cb94-4"><a href="cancer-dependency-map-depmap.html#cb94-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   depmap_id  lineage gene   rna_expression entrez_id gene_name cell_line     </span></span>
<span id="cb94-5"><a href="cancer-dependency-map-depmap.html#cb94-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;      &lt;chr&gt;   &lt;chr&gt;           &lt;dbl&gt;     &lt;int&gt; &lt;chr&gt;     &lt;chr&gt;         </span></span>
<span id="cb94-6"><a href="cancer-dependency-map-depmap.html#cb94-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 ACH-000001 ovary   GATA6…         1.27        2627 GATA6     NIHOVCAR3_OVA…</span></span>
<span id="cb94-7"><a href="cancer-dependency-map-depmap.html#cb94-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 ACH-000002 blood   GATA6…         0           2627 GATA6     HL60_HAEMATOP…</span></span>
<span id="cb94-8"><a href="cancer-dependency-map-depmap.html#cb94-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3 ACH-000003 colore… GATA6…         5.91        2627 GATA6     CACO2_LARGE_I…</span></span>
<span id="cb94-9"><a href="cancer-dependency-map-depmap.html#cb94-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4 ACH-000004 blood   GATA6…         0.0704      2627 GATA6     HEL_HAEMATOPO…</span></span>
<span id="cb94-10"><a href="cancer-dependency-map-depmap.html#cb94-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5 ACH-000005 blood   GATA6…         0.202       2627 GATA6     HEL9217_HAEMA…</span></span>
<span id="cb94-11"><a href="cancer-dependency-map-depmap.html#cb94-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6 ACH-000006 blood   GATA6…         0           2627 GATA6     MONOMAC6_HAEM…</span></span></code></pre></div>
<p>Using the descriptive plotting skills that you learned in <strong>Chapter 4</strong> generate a boxplot to determine whether GATA6 gene expression is specifically enriched in a cell line lineage.</p>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="cancer-dependency-map-depmap.html#cb95-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb95-2"><a href="cancer-dependency-map-depmap.html#cb95-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggboxplot</span>(gata6Bylineage, <span class="at">x =</span> <span class="st">&quot;lineage&quot;</span>, <span class="at">y =</span> <span class="st">&quot;rna_expression&quot;</span>,</span>
<span id="cb95-3"><a href="cancer-dependency-map-depmap.html#cb95-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">title =</span> <span class="st">&quot;GATA6 Expression by Cell Line Lineage&quot;</span>, <span class="at">ylab =</span> <span class="st">&quot;Expression (TPM)&quot;</span>,</span>
<span id="cb95-4"><a href="cancer-dependency-map-depmap.html#cb95-4" aria-hidden="true" tabindex="-1"></a>          <span class="at">color =</span> <span class="st">&quot;lineage&quot;</span>) <span class="sc">+</span> <span class="fu">rotate_x_text</span>()</span></code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-100"></span>
<img src="PGT_Bioinformatics_course_files/figure-html/unnamed-chunk-100-1.png" alt="Boxplot of GATA6 expression (TPM) by cell lineage" width="960" />
<p class="caption">
Figure 7.1: Boxplot of GATA6 expression (TPM) by cell lineage
</p>
</div>
<p>We might also be interested in the relative expression of GATA6 in different Pancreatic Cancer cell lines. We can generate a barplot to show the relative expression levels as follows:</p>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="cancer-dependency-map-depmap.html#cb96-1" aria-hidden="true" tabindex="-1"></a>gata6BylPancreas <span class="ot">&lt;-</span> gata6Bylineage <span class="sc">%&gt;%</span> <span class="fu">filter</span>(lineage<span class="sc">==</span><span class="st">&quot;pancreas&quot;</span>)</span>
<span id="cb96-2"><a href="cancer-dependency-map-depmap.html#cb96-2" aria-hidden="true" tabindex="-1"></a>gata6BylPancreas <span class="sc">%&gt;%</span> <span class="fu">head</span>()</span>
<span id="cb96-3"><a href="cancer-dependency-map-depmap.html#cb96-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 6 x 7</span></span>
<span id="cb96-4"><a href="cancer-dependency-map-depmap.html#cb96-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   depmap_id  lineage  gene    rna_expression entrez_id gene_name cell_line   </span></span>
<span id="cb96-5"><a href="cancer-dependency-map-depmap.html#cb96-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;      &lt;chr&gt;    &lt;chr&gt;            &lt;dbl&gt;     &lt;int&gt; &lt;chr&gt;     &lt;chr&gt;       </span></span>
<span id="cb96-6"><a href="cancer-dependency-map-depmap.html#cb96-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 ACH-000022 pancreas GATA6 …          5.23       2627 GATA6     PATU8988S_P…</span></span>
<span id="cb96-7"><a href="cancer-dependency-map-depmap.html#cb96-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 ACH-000023 pancreas GATA6 …          0.333      2627 GATA6     PATU8988T_P…</span></span>
<span id="cb96-8"><a href="cancer-dependency-map-depmap.html#cb96-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3 ACH-000031 pancreas GATA6 …          2.99       2627 GATA6     PANC0213_PA…</span></span>
<span id="cb96-9"><a href="cancer-dependency-map-depmap.html#cb96-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4 ACH-000042 pancreas GATA6 …          1.03       2627 GATA6     PANC0203_PA…</span></span>
<span id="cb96-10"><a href="cancer-dependency-map-depmap.html#cb96-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5 ACH-000060 pancreas GATA6 …          0.310      2627 GATA6     PANC1005_PA…</span></span>
<span id="cb96-11"><a href="cancer-dependency-map-depmap.html#cb96-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6 ACH-000085 pancreas GATA6 …          2.74       2627 GATA6     T3M4_PANCRE…</span></span></code></pre></div>
<p>Generate a barplot using the <code>ggbarplot</code> function with expression values plotted in descending order:</p>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="cancer-dependency-map-depmap.html#cb97-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggbarplot</span>(gata6BylPancreas, <span class="at">x =</span> <span class="st">&quot;cell_line&quot;</span>, <span class="at">y=</span><span class="st">&quot;rna_expression&quot;</span>, </span>
<span id="cb97-2"><a href="cancer-dependency-map-depmap.html#cb97-2" aria-hidden="true" tabindex="-1"></a>          <span class="at">fill =</span> <span class="st">&quot;seashell3&quot;</span>, <span class="at">sort.val =</span> <span class="st">&quot;desc&quot;</span>,</span>
<span id="cb97-3"><a href="cancer-dependency-map-depmap.html#cb97-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">title =</span> <span class="st">&quot;GATA6 Expression in Pancreatic Cancer Cell lines&quot;</span>, <span class="at">ylab =</span> <span class="st">&quot;Expression (TPM)&quot;</span>,</span>
<span id="cb97-4"><a href="cancer-dependency-map-depmap.html#cb97-4" aria-hidden="true" tabindex="-1"></a>          <span class="at">xlab =</span> <span class="st">&quot;Pancreatic Cancer Cell Lines &quot;</span>) <span class="sc">+</span> <span class="fu">rotate_x_text</span>()</span></code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-102"></span>
<img src="PGT_Bioinformatics_course_files/figure-html/unnamed-chunk-102-1.png" alt="Barplot of GATA6 expression (TPM) in Pancreatic Cancer Cell lines" width="960" />
<p class="caption">
Figure 7.2: Barplot of GATA6 expression (TPM) in Pancreatic Cancer Cell lines
</p>
</div>
</div>
<div id="exploring-depmap-mutation-calls" class="section level2" number="7.3">
<h2><span class="header-section-number">7.3</span> Exploring DepMap mutation calls</h2>
<p>First, let’s extract the 21Q2 DepMap mutation calls:</p>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="cancer-dependency-map-depmap.html#cb98-1" aria-hidden="true" tabindex="-1"></a>mutData <span class="ot">&lt;-</span> eh[[<span class="st">&quot;EH6121&quot;</span>]]</span>
<span id="cb98-2"><a href="cancer-dependency-map-depmap.html#cb98-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; see ?depmap and browseVignettes(&#39;depmap&#39;) for documentation</span></span>
<span id="cb98-3"><a href="cancer-dependency-map-depmap.html#cb98-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; loading from cache</span></span></code></pre></div>
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="cancer-dependency-map-depmap.html#cb99-1" aria-hidden="true" tabindex="-1"></a>mutBylineage <span class="ot">&lt;-</span> metaData <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(depmap_id, lineage, cell_line_name) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">full_join</span>(mutData, <span class="at">by =</span> <span class="st">&quot;depmap_id&quot;</span>) </span>
<span id="cb99-2"><a href="cancer-dependency-map-depmap.html#cb99-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(mutBylineage)</span>
<span id="cb99-3"><a href="cancer-dependency-map-depmap.html#cb99-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 6 x 34</span></span>
<span id="cb99-4"><a href="cancer-dependency-map-depmap.html#cb99-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   depmap_id  lineage cell_line_name gene_name entrez_id ncbi_build chromosome</span></span>
<span id="cb99-5"><a href="cancer-dependency-map-depmap.html#cb99-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;      &lt;chr&gt;   &lt;chr&gt;          &lt;chr&gt;         &lt;dbl&gt;      &lt;dbl&gt; &lt;chr&gt;     </span></span>
<span id="cb99-6"><a href="cancer-dependency-map-depmap.html#cb99-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 ACH-000001 ovary   NIH:OVCAR-3    VPS13D        55187         37 1         </span></span>
<span id="cb99-7"><a href="cancer-dependency-map-depmap.html#cb99-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 ACH-000001 ovary   NIH:OVCAR-3    AADACL4      343066         37 1         </span></span>
<span id="cb99-8"><a href="cancer-dependency-map-depmap.html#cb99-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3 ACH-000001 ovary   NIH:OVCAR-3    IFNLR1       163702         37 1         </span></span>
<span id="cb99-9"><a href="cancer-dependency-map-depmap.html#cb99-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4 ACH-000001 ovary   NIH:OVCAR-3    TMEM57        55219         37 1         </span></span>
<span id="cb99-10"><a href="cancer-dependency-map-depmap.html#cb99-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5 ACH-000001 ovary   NIH:OVCAR-3    ZSCAN20        7579         37 1         </span></span>
<span id="cb99-11"><a href="cancer-dependency-map-depmap.html#cb99-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6 ACH-000001 ovary   NIH:OVCAR-3    POU3F1         5453         37 1         </span></span>
<span id="cb99-12"><a href="cancer-dependency-map-depmap.html#cb99-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # … with 27 more variables: start_pos &lt;dbl&gt;, end_pos &lt;dbl&gt;, strand &lt;chr&gt;,</span></span>
<span id="cb99-13"><a href="cancer-dependency-map-depmap.html#cb99-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   var_class &lt;chr&gt;, var_type &lt;chr&gt;, ref_allele &lt;chr&gt;,</span></span>
<span id="cb99-14"><a href="cancer-dependency-map-depmap.html#cb99-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   tumor_seq_allele1 &lt;chr&gt;, dbSNP_RS &lt;chr&gt;, dbSNP_val_status &lt;chr&gt;,</span></span>
<span id="cb99-15"><a href="cancer-dependency-map-depmap.html#cb99-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   genome_change &lt;chr&gt;, annotation_trans &lt;chr&gt;, cDNA_change &lt;chr&gt;,</span></span>
<span id="cb99-16"><a href="cancer-dependency-map-depmap.html#cb99-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   codon_change &lt;chr&gt;, protein_change &lt;chr&gt;, is_deleterious &lt;lgl&gt;,</span></span>
<span id="cb99-17"><a href="cancer-dependency-map-depmap.html#cb99-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   is_tcga_hotspot &lt;lgl&gt;, tcga_hsCnt &lt;dbl&gt;, is_cosmic_hotspot &lt;lgl&gt;,</span></span>
<span id="cb99-18"><a href="cancer-dependency-map-depmap.html#cb99-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   cosmic_hsCnt &lt;dbl&gt;, ExAC_AF &lt;dbl&gt;, var_annotation &lt;chr&gt;,</span></span>
<span id="cb99-19"><a href="cancer-dependency-map-depmap.html#cb99-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   CGA_WES_AC &lt;chr&gt;, HC_AC &lt;chr&gt;, RD_AC &lt;chr&gt;, RNAseq_AC &lt;chr&gt;,</span></span>
<span id="cb99-20"><a href="cancer-dependency-map-depmap.html#cb99-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; #   sanger_WES_AC &lt;chr&gt;, WGS_AC &lt;chr&gt;</span></span></code></pre></div>
<p>The <code>maftools</code> package provides a number of useful functions for exploring and analyzing mutation calls. Before we can use maftools we need to modify some of the column names.</p>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="cancer-dependency-map-depmap.html#cb100-1" aria-hidden="true" tabindex="-1"></a>mutBylineageRename <span class="ot">&lt;-</span> mutBylineage <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">Hugo_Symbol=</span>gene_name) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">Start_Position=</span>start_pos) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">End_Position=</span>end_pos) <span class="sc">%&gt;%</span></span>
<span id="cb100-2"><a href="cancer-dependency-map-depmap.html#cb100-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Variant_Classification=</span>var_class) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">Variant_Type=</span>var_type) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">Reference_Allele=</span>ref_allele) <span class="sc">%&gt;%</span> </span>
<span id="cb100-3"><a href="cancer-dependency-map-depmap.html#cb100-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Tumor_Seq_Allele2 =</span> tumor_seq_allele1) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">Tumor_Sample_Barcode =</span> cell_line_name)</span></code></pre></div>
<p>Create a mafObject using our renamed table:</p>
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="cancer-dependency-map-depmap.html#cb101-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(maftools)</span>
<span id="cb101-2"><a href="cancer-dependency-map-depmap.html#cb101-2" aria-hidden="true" tabindex="-1"></a>mafObject <span class="ot">&lt;-</span> <span class="fu">read.maf</span>(mutBylineageRename, <span class="at">rmFlags =</span> <span class="cn">TRUE</span>)</span>
<span id="cb101-3"><a href="cancer-dependency-map-depmap.html#cb101-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; -Validating</span></span>
<span id="cb101-4"><a href="cancer-dependency-map-depmap.html#cb101-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; --Removed 567 duplicated variants</span></span>
<span id="cb101-5"><a href="cancer-dependency-map-depmap.html#cb101-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; --Found  46  variants with no Gene Symbols</span></span>
<span id="cb101-6"><a href="cancer-dependency-map-depmap.html#cb101-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; --Annotating them as &#39;UnknownGene&#39; for convenience</span></span>
<span id="cb101-7"><a href="cancer-dependency-map-depmap.html#cb101-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; --Non MAF specific values in Variant_Classification column:</span></span>
<span id="cb101-8"><a href="cancer-dependency-map-depmap.html#cb101-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   Start_Codon_Del</span></span>
<span id="cb101-9"><a href="cancer-dependency-map-depmap.html#cb101-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   Stop_Codon_Ins</span></span>
<span id="cb101-10"><a href="cancer-dependency-map-depmap.html#cb101-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   NA</span></span>
<span id="cb101-11"><a href="cancer-dependency-map-depmap.html#cb101-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; --Non MAF specific values in Variant_Type column:</span></span>
<span id="cb101-12"><a href="cancer-dependency-map-depmap.html#cb101-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   NA</span></span>
<span id="cb101-13"><a href="cancer-dependency-map-depmap.html#cb101-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; -Removing 20 FLAG genes</span></span>
<span id="cb101-14"><a href="cancer-dependency-map-depmap.html#cb101-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; -Silent variants: 397886 </span></span>
<span id="cb101-15"><a href="cancer-dependency-map-depmap.html#cb101-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; -Summarizing</span></span>
<span id="cb101-16"><a href="cancer-dependency-map-depmap.html#cb101-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; -Processing clinical data</span></span>
<span id="cb101-17"><a href="cancer-dependency-map-depmap.html#cb101-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; --Missing clinical data</span></span>
<span id="cb101-18"><a href="cancer-dependency-map-depmap.html#cb101-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; -Finished in 32.7s elapsed (28.5s cpu)</span></span></code></pre></div>
<p>Subset the mafObject to select only pancreatic cell lines and genes that are know pancreatic cancer drivers.</p>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="cancer-dependency-map-depmap.html#cb102-1" aria-hidden="true" tabindex="-1"></a>drivers <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;TP53,SMAD4,KRAS,RNF43,ARID1A,CDKN2A,TGFBR2,ACVR1B,KDM6A,MLL3,ACVR2A,SMARCA4,MAP2K4,SMAD3,RBM10,SF3B1,MARK2,TGFBR1,SDK2,PBRM1,U2AF1,FBXW7,ATM,TP53BP2,CALD1,BCORL1,FBLN2,PLXNB2,SIN3B,MACF1,DISP2,KALRN,ITPR3,HIVEP1,MYCBP2,NF2,SETD2,SPTB,NBEA,TLE4,JAG1,BRCA1,BRCA2,RPA1,PALB2&quot;</span>)</span>
<span id="cb102-2"><a href="cancer-dependency-map-depmap.html#cb102-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-3"><a href="cancer-dependency-map-depmap.html#cb102-3" aria-hidden="true" tabindex="-1"></a>drivers <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(drivers, <span class="st">&quot;,&quot;</span>)[[<span class="dv">1</span>]])</span>
<span id="cb102-4"><a href="cancer-dependency-map-depmap.html#cb102-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-5"><a href="cancer-dependency-map-depmap.html#cb102-5" aria-hidden="true" tabindex="-1"></a>mafPanc <span class="ot">&lt;-</span> <span class="fu">subsetMaf</span>(mafObject, <span class="at">query=</span><span class="st">&quot;lineage == &#39;pancreas&#39;&quot;</span>, <span class="at">genes =</span> drivers)</span>
<span id="cb102-6"><a href="cancer-dependency-map-depmap.html#cb102-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; -Processing clinical data</span></span>
<span id="cb102-7"><a href="cancer-dependency-map-depmap.html#cb102-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-8"><a href="cancer-dependency-map-depmap.html#cb102-8" aria-hidden="true" tabindex="-1"></a>mafPanc</span>
<span id="cb102-9"><a href="cancer-dependency-map-depmap.html#cb102-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; An object of class  MAF </span></span>
<span id="cb102-10"><a href="cancer-dependency-map-depmap.html#cb102-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                    ID summary  Mean Median</span></span>
<span id="cb102-11"><a href="cancer-dependency-map-depmap.html#cb102-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  1:        NCBI_Build      NA    NA     NA</span></span>
<span id="cb102-12"><a href="cancer-dependency-map-depmap.html#cb102-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  2:            Center      NA    NA     NA</span></span>
<span id="cb102-13"><a href="cancer-dependency-map-depmap.html#cb102-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  3:           Samples      56    NA     NA</span></span>
<span id="cb102-14"><a href="cancer-dependency-map-depmap.html#cb102-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  4:            nGenes      37    NA     NA</span></span>
<span id="cb102-15"><a href="cancer-dependency-map-depmap.html#cb102-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  5:   Frame_Shift_Del      25 0.446    0.0</span></span>
<span id="cb102-16"><a href="cancer-dependency-map-depmap.html#cb102-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  6:   Frame_Shift_Ins      24 0.429    0.0</span></span>
<span id="cb102-17"><a href="cancer-dependency-map-depmap.html#cb102-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  7:      In_Frame_Del       4 0.071    0.0</span></span>
<span id="cb102-18"><a href="cancer-dependency-map-depmap.html#cb102-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  8:      In_Frame_Ins       2 0.036    0.0</span></span>
<span id="cb102-19"><a href="cancer-dependency-map-depmap.html#cb102-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  9: Missense_Mutation     156 2.786    2.5</span></span>
<span id="cb102-20"><a href="cancer-dependency-map-depmap.html#cb102-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10: Nonsense_Mutation      23 0.411    0.0</span></span>
<span id="cb102-21"><a href="cancer-dependency-map-depmap.html#cb102-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 11:       Splice_Site      12 0.214    0.0</span></span>
<span id="cb102-22"><a href="cancer-dependency-map-depmap.html#cb102-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 12:             total     246 4.393    4.0</span></span></code></pre></div>
<p>Plot a summary of the mutation calls:</p>
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="cancer-dependency-map-depmap.html#cb103-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotmafSummary</span>(<span class="at">maf =</span> mafPanc, <span class="at">rmOutlier =</span> <span class="cn">TRUE</span>, <span class="at">addStat =</span> <span class="st">&#39;median&#39;</span>, <span class="at">dashboard =</span> <span class="cn">TRUE</span>, <span class="at">titvRaw =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-108"></span>
<img src="PGT_Bioinformatics_course_files/figure-html/unnamed-chunk-108-1.png" alt="Summary of mutations found in Pancreatic Cancer Cell Lines" width="960" />
<p class="caption">
Figure 7.3: Summary of mutations found in Pancreatic Cancer Cell Lines
</p>
</div>
<p>Generate an oncoplot of the top 20 mutations:</p>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="cancer-dependency-map-depmap.html#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="fu">oncoplot</span>(<span class="at">maf =</span> mafPanc, <span class="at">top =</span> <span class="dv">20</span>, <span class="at">showTumorSampleBarcodes =</span> <span class="cn">TRUE</span>, <span class="at">SampleNamefontSize =</span> <span class="fl">0.5</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-109"></span>
<img src="PGT_Bioinformatics_course_files/figure-html/unnamed-chunk-109-1.png" alt="Summary of mutations found in Pancreatic Cancer Cell Lines" width="960" />
<p class="caption">
Figure 7.4: Summary of mutations found in Pancreatic Cancer Cell Lines
</p>
</div>
</div>
<div id="exploring-depmap-crispr-data" class="section level2" number="7.4">
<h2><span class="header-section-number">7.4</span> Exploring DepMap CRISPR data</h2>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="cancer-dependency-map-depmap.html#cb105-1" aria-hidden="true" tabindex="-1"></a>crispData <span class="ot">&lt;-</span> eh[[<span class="st">&quot;EH6118&quot;</span>]]</span>
<span id="cb105-2"><a href="cancer-dependency-map-depmap.html#cb105-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; see ?depmap and browseVignettes(&#39;depmap&#39;) for documentation</span></span>
<span id="cb105-3"><a href="cancer-dependency-map-depmap.html#cb105-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; loading from cache</span></span>
<span id="cb105-4"><a href="cancer-dependency-map-depmap.html#cb105-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(crispData)</span>
<span id="cb105-5"><a href="cancer-dependency-map-depmap.html#cb105-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 6 x 6</span></span>
<span id="cb105-6"><a href="cancer-dependency-map-depmap.html#cb105-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   depmap_id  gene    dependency entrez_id gene_name cell_line                </span></span>
<span id="cb105-7"><a href="cancer-dependency-map-depmap.html#cb105-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;      &lt;chr&gt;        &lt;dbl&gt;     &lt;int&gt; &lt;chr&gt;     &lt;chr&gt;                    </span></span>
<span id="cb105-8"><a href="cancer-dependency-map-depmap.html#cb105-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 ACH-000001 A1BG (…   -0.335           1 A1BG      NIHOVCAR3_OVARY          </span></span>
<span id="cb105-9"><a href="cancer-dependency-map-depmap.html#cb105-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 ACH-000004 A1BG (…    0.0201          1 A1BG      HEL_HAEMATOPOIETIC_AND_L…</span></span>
<span id="cb105-10"><a href="cancer-dependency-map-depmap.html#cb105-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3 ACH-000005 A1BG (…   -0.191           1 A1BG      HEL9217_HAEMATOPOIETIC_A…</span></span>
<span id="cb105-11"><a href="cancer-dependency-map-depmap.html#cb105-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4 ACH-000007 A1BG (…    0.00886         1 A1BG      LS513_LARGE_INTESTINE    </span></span>
<span id="cb105-12"><a href="cancer-dependency-map-depmap.html#cb105-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5 ACH-000009 A1BG (…    0.00648         1 A1BG      C2BBE1_LARGE_INTESTINE   </span></span>
<span id="cb105-13"><a href="cancer-dependency-map-depmap.html#cb105-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6 ACH-000011 A1BG (…    0.145           1 A1BG      253J_URINARY_TRACT</span></span></code></pre></div>
<p>Join with metadata.</p>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="cancer-dependency-map-depmap.html#cb106-1" aria-hidden="true" tabindex="-1"></a>crispBylineage <span class="ot">&lt;-</span> metaData <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(depmap_id, lineage) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">full_join</span>(crispData, <span class="at">by =</span> <span class="st">&quot;depmap_id&quot;</span>) </span>
<span id="cb106-2"><a href="cancer-dependency-map-depmap.html#cb106-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(crispBylineage)</span>
<span id="cb106-3"><a href="cancer-dependency-map-depmap.html#cb106-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 6 x 7</span></span>
<span id="cb106-4"><a href="cancer-dependency-map-depmap.html#cb106-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   depmap_id  lineage gene         dependency entrez_id gene_name cell_line   </span></span>
<span id="cb106-5"><a href="cancer-dependency-map-depmap.html#cb106-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;      &lt;chr&gt;   &lt;chr&gt;             &lt;dbl&gt;     &lt;int&gt; &lt;chr&gt;     &lt;chr&gt;       </span></span>
<span id="cb106-6"><a href="cancer-dependency-map-depmap.html#cb106-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 ACH-000001 ovary   A1BG (1)        -0.335          1 A1BG      NIHOVCAR3_O…</span></span>
<span id="cb106-7"><a href="cancer-dependency-map-depmap.html#cb106-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 ACH-000001 ovary   A1CF (29974)    -0.0616     29974 A1CF      NIHOVCAR3_O…</span></span>
<span id="cb106-8"><a href="cancer-dependency-map-depmap.html#cb106-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3 ACH-000001 ovary   A2M (2)         -0.0269         2 A2M       NIHOVCAR3_O…</span></span>
<span id="cb106-9"><a href="cancer-dependency-map-depmap.html#cb106-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4 ACH-000001 ovary   A2ML1 (1445…    -0.0265    144568 A2ML1     NIHOVCAR3_O…</span></span>
<span id="cb106-10"><a href="cancer-dependency-map-depmap.html#cb106-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5 ACH-000001 ovary   A3GALT2 (12…    -0.130     127550 A3GALT2   NIHOVCAR3_O…</span></span>
<span id="cb106-11"><a href="cancer-dependency-map-depmap.html#cb106-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6 ACH-000001 ovary   A4GALT (539…     0.0607     53947 A4GALT    NIHOVCAR3_O…</span></span></code></pre></div>
<p>Filter to select GATA6 and Pancreatic Cancer Cell Lines</p>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="cancer-dependency-map-depmap.html#cb107-1" aria-hidden="true" tabindex="-1"></a>gata6CrispPanc <span class="ot">&lt;-</span> crispBylineage <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(gene_name <span class="sc">==</span> <span class="st">&quot;GATA6&quot;</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(lineage <span class="sc">==</span> <span class="st">&quot;pancreas&quot;</span>)</span>
<span id="cb107-2"><a href="cancer-dependency-map-depmap.html#cb107-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(gata6CrispPanc)</span>
<span id="cb107-3"><a href="cancer-dependency-map-depmap.html#cb107-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # A tibble: 6 x 7</span></span>
<span id="cb107-4"><a href="cancer-dependency-map-depmap.html#cb107-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   depmap_id  lineage  gene      dependency entrez_id gene_name cell_line     </span></span>
<span id="cb107-5"><a href="cancer-dependency-map-depmap.html#cb107-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;      &lt;chr&gt;    &lt;chr&gt;          &lt;dbl&gt;     &lt;int&gt; &lt;chr&gt;     &lt;chr&gt;         </span></span>
<span id="cb107-6"><a href="cancer-dependency-map-depmap.html#cb107-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 ACH-000022 pancreas GATA6 (2…   -0.476        2627 GATA6     PATU8988S_PAN…</span></span>
<span id="cb107-7"><a href="cancer-dependency-map-depmap.html#cb107-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 ACH-000023 pancreas GATA6 (2…    0.210        2627 GATA6     PATU8988T_PAN…</span></span>
<span id="cb107-8"><a href="cancer-dependency-map-depmap.html#cb107-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3 ACH-000042 pancreas GATA6 (2…   -0.0438       2627 GATA6     PANC0203_PANC…</span></span>
<span id="cb107-9"><a href="cancer-dependency-map-depmap.html#cb107-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4 ACH-000060 pancreas GATA6 (2…   -0.0412       2627 GATA6     PANC1005_PANC…</span></span>
<span id="cb107-10"><a href="cancer-dependency-map-depmap.html#cb107-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5 ACH-000085 pancreas GATA6 (2…   -0.0496       2627 GATA6     T3M4_PANCREAS </span></span>
<span id="cb107-11"><a href="cancer-dependency-map-depmap.html#cb107-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6 ACH-000093 pancreas GATA6 (2…    0.00316      2627 GATA6     PANC0504_PANC…</span></span></code></pre></div>
<p>Join GATA6 Expression and GATA6 CRISPRi tables to obtain final table.</p>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="cancer-dependency-map-depmap.html#cb108-1" aria-hidden="true" tabindex="-1"></a>gata6CrispExpPanc <span class="ot">&lt;-</span> gata6CrispPanc <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">full_join</span>(gata6BylPancreas, <span class="at">by =</span> <span class="st">&quot;depmap_id&quot;</span>) </span></code></pre></div>
<p>Generate a Scatter Plot using the <code>ggscatter</code> function with added labels.</p>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="cancer-dependency-map-depmap.html#cb109-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggscatter</span>(gata6CrispExpPanc, <span class="at">x =</span> <span class="st">&quot;dependency&quot;</span>, <span class="at">y =</span> <span class="st">&quot;rna_expression&quot;</span>, <span class="at">size =</span> <span class="fl">0.6</span>, </span>
<span id="cb109-2"><a href="cancer-dependency-map-depmap.html#cb109-2" aria-hidden="true" tabindex="-1"></a>          <span class="at">rug =</span> <span class="cn">TRUE</span>, <span class="co"># Add marginal rug</span></span>
<span id="cb109-3"><a href="cancer-dependency-map-depmap.html#cb109-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">add =</span> <span class="st">&quot;reg.line&quot;</span>,  <span class="at">label =</span> <span class="st">&quot;cell_line.x&quot;</span>, </span>
<span id="cb109-4"><a href="cancer-dependency-map-depmap.html#cb109-4" aria-hidden="true" tabindex="-1"></a>          <span class="at">repel =</span> <span class="cn">TRUE</span>,</span>
<span id="cb109-5"><a href="cancer-dependency-map-depmap.html#cb109-5" aria-hidden="true" tabindex="-1"></a>          <span class="at">font.label =</span> <span class="fu">list</span>(<span class="at">size =</span> <span class="dv">9</span>, <span class="at">face =</span> <span class="st">&quot;italic&quot;</span>),</span>
<span id="cb109-6"><a href="cancer-dependency-map-depmap.html#cb109-6" aria-hidden="true" tabindex="-1"></a>          <span class="at">ylab =</span> <span class="st">&quot;GATA6 Expression (TPM)&quot;</span>,</span>
<span id="cb109-7"><a href="cancer-dependency-map-depmap.html#cb109-7" aria-hidden="true" tabindex="-1"></a>          <span class="at">xlab =</span> <span class="st">&quot;GATA6 CRISPRi Dependency&quot;</span></span>
<span id="cb109-8"><a href="cancer-dependency-map-depmap.html#cb109-8" aria-hidden="true" tabindex="-1"></a>          )</span></code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-114"></span>
<img src="PGT_Bioinformatics_course_files/figure-html/unnamed-chunk-114-1.png" alt="GATA6 Expression versus CRISPRi Dependency in Pancreatic Cancer Cell Lines" width="480" />
<p class="caption">
Figure 7.5: GATA6 Expression versus CRISPRi Dependency in Pancreatic Cancer Cell Lines
</p>
</div>
<p>Generate a Scatter Plot using the <code>ggscatter</code> function showing only those cell lines that have high GATA6 gene expression and GATA6 dependency. Let’s also determine the significance of the correlation.</p>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="cancer-dependency-map-depmap.html#cb110-1" aria-hidden="true" tabindex="-1"></a>label.select.criteria <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">criteria =</span> <span class="st">&quot;`dependency` &lt; -0.2 &amp; `rna_expression` &gt;= 3&quot;</span>)</span>
<span id="cb110-2"><a href="cancer-dependency-map-depmap.html#cb110-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-3"><a href="cancer-dependency-map-depmap.html#cb110-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggscatter</span>(gata6CrispExpPanc, <span class="at">x =</span> <span class="st">&quot;dependency&quot;</span>, <span class="at">y =</span> <span class="st">&quot;rna_expression&quot;</span>, <span class="at">size =</span> <span class="fl">0.6</span>, </span>
<span id="cb110-4"><a href="cancer-dependency-map-depmap.html#cb110-4" aria-hidden="true" tabindex="-1"></a>          <span class="at">rug =</span> <span class="cn">TRUE</span>,</span>
<span id="cb110-5"><a href="cancer-dependency-map-depmap.html#cb110-5" aria-hidden="true" tabindex="-1"></a>          <span class="at">label.select =</span> label.select.criteria, </span>
<span id="cb110-6"><a href="cancer-dependency-map-depmap.html#cb110-6" aria-hidden="true" tabindex="-1"></a>          <span class="at">font.label =</span> <span class="fu">list</span>(<span class="at">size =</span> <span class="dv">9</span>, <span class="at">face =</span> <span class="st">&quot;italic&quot;</span>),</span>
<span id="cb110-7"><a href="cancer-dependency-map-depmap.html#cb110-7" aria-hidden="true" tabindex="-1"></a>          <span class="at">add =</span> <span class="st">&quot;reg.line&quot;</span>,  <span class="at">label =</span> <span class="st">&quot;cell_line.x&quot;</span>, <span class="at">repel =</span> <span class="cn">TRUE</span>,</span>
<span id="cb110-8"><a href="cancer-dependency-map-depmap.html#cb110-8" aria-hidden="true" tabindex="-1"></a>          <span class="at">cor.coef =</span> <span class="cn">TRUE</span>, </span>
<span id="cb110-9"><a href="cancer-dependency-map-depmap.html#cb110-9" aria-hidden="true" tabindex="-1"></a>          <span class="at">cor.coeff.args =</span> <span class="fu">list</span>(<span class="at">method =</span> <span class="st">&quot;pearson&quot;</span>, <span class="at">label.x =</span> <span class="fl">0.25</span>, <span class="at">label.sep =</span> <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>),</span>
<span id="cb110-10"><a href="cancer-dependency-map-depmap.html#cb110-10" aria-hidden="true" tabindex="-1"></a>          <span class="at">ylab =</span> <span class="st">&quot;GATA6 Expression (TPM)&quot;</span>,</span>
<span id="cb110-11"><a href="cancer-dependency-map-depmap.html#cb110-11" aria-hidden="true" tabindex="-1"></a>          <span class="at">xlab =</span> <span class="st">&quot;GATA6 CRISPRi Dependency&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:unnamed-chunk-115"></span>
<img src="PGT_Bioinformatics_course_files/figure-html/unnamed-chunk-115-1.png" alt="GATA6 Expression versus CRISPRi Dependency in Pancreatic Cancer Cell Lines" width="480" />
<p class="caption">
Figure 7.6: GATA6 Expression versus CRISPRi Dependency in Pancreatic Cancer Cell Lines
</p>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="rnaseq-analysis-deseq2.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["PGT_Bioinformatics_course.pdf", "PGT_Bioinformatics_course.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
