<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 RNAseq Analysis | PGT Bioinformatics Course</title>
  <meta name="description" content="Basic introduction to RNAseq analysis using R/Bioconductor" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 RNAseq Analysis | PGT Bioinformatics Course" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Basic introduction to RNAseq analysis using R/Bioconductor" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 RNAseq Analysis | PGT Bioinformatics Course" />
  
  <meta name="twitter:description" content="Basic introduction to RNAseq analysis using R/Bioconductor" />
  

<meta name="author" content="Peter J. Bailey" />


<meta name="date" content="2021-07-19" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="descriptive-statistics-and-plotting.html"/>
<link rel="next" href="rnaseq-analysis-deseq2.html"/>
<script src="libs/header-attrs-2.9/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">PGT Bioinformatics Course</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#installing-r-and-rstudio"><i class="fa fa-check"></i><b>1.1</b> Installing R and RStudio</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#installing-r-and-bioconductor-packages"><i class="fa fa-check"></i><b>1.2</b> Installing R and Bioconductor packages</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#loading-r-and-bioconductor-packages"><i class="fa fa-check"></i><b>1.3</b> Loading R and Bioconductor packages</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#learning_resources"><i class="fa fa-check"></i><b>1.4</b> Learning resources</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> Introduction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="intro.html"><a href="intro.html#course-outline"><i class="fa fa-check"></i><b>2.1</b> Course outline</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="intro.html"><a href="intro.html#part1"><i class="fa fa-check"></i><b>2.1.1</b> PART 1</a></li>
<li class="chapter" data-level="2.1.2" data-path="intro.html"><a href="intro.html#part2"><i class="fa fa-check"></i><b>2.1.2</b> PART 2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="r-programming-language-primer.html"><a href="r-programming-language-primer.html"><i class="fa fa-check"></i><b>3</b> R Programming language primer</a>
<ul>
<li class="chapter" data-level="3.1" data-path="r-programming-language-primer.html"><a href="r-programming-language-primer.html#vectors"><i class="fa fa-check"></i><b>3.1</b> Vectors</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="r-programming-language-primer.html"><a href="r-programming-language-primer.html#vectors-in-action"><i class="fa fa-check"></i><b>3.1.1</b> Vectors in action</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="r-programming-language-primer.html"><a href="r-programming-language-primer.html#matrix"><i class="fa fa-check"></i><b>3.2</b> Matrix</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="r-programming-language-primer.html"><a href="r-programming-language-primer.html#matrices-in-action"><i class="fa fa-check"></i><b>3.2.1</b> Matrices in action</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="r-programming-language-primer.html"><a href="r-programming-language-primer.html#data-frame"><i class="fa fa-check"></i><b>3.3</b> Data Frame</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="r-programming-language-primer.html"><a href="r-programming-language-primer.html#data-frames-in-action"><i class="fa fa-check"></i><b>3.3.1</b> Data Frames in action</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="r-programming-language-primer.html"><a href="r-programming-language-primer.html#factors"><i class="fa fa-check"></i><b>3.4</b> Factors</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="r-programming-language-primer.html"><a href="r-programming-language-primer.html#factors-in-action"><i class="fa fa-check"></i><b>3.4.1</b> Factors in action</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="r-programming-language-primer.html"><a href="r-programming-language-primer.html#list"><i class="fa fa-check"></i><b>3.5</b> List</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="r-programming-language-primer.html"><a href="r-programming-language-primer.html#lists-in-action"><i class="fa fa-check"></i><b>3.5.1</b> Lists in action</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="r-programming-language-primer.html"><a href="r-programming-language-primer.html#array"><i class="fa fa-check"></i><b>3.6</b> Array</a></li>
<li class="chapter" data-level="3.7" data-path="r-programming-language-primer.html"><a href="r-programming-language-primer.html#creating-functions"><i class="fa fa-check"></i><b>3.7</b> Creating functions</a></li>
<li class="chapter" data-level="3.8" data-path="r-programming-language-primer.html"><a href="r-programming-language-primer.html#loading-data"><i class="fa fa-check"></i><b>3.8</b> Loading Data</a></li>
<li class="chapter" data-level="3.9" data-path="r-programming-language-primer.html"><a href="r-programming-language-primer.html#saving-data"><i class="fa fa-check"></i><b>3.9</b> Saving Data</a></li>
<li class="chapter" data-level="3.10" data-path="r-programming-language-primer.html"><a href="r-programming-language-primer.html#saving-plots"><i class="fa fa-check"></i><b>3.10</b> Saving plots</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="descriptive-statistics-and-plotting.html"><a href="descriptive-statistics-and-plotting.html"><i class="fa fa-check"></i><b>4</b> Descriptive statistics and plotting</a>
<ul>
<li class="chapter" data-level="4.1" data-path="descriptive-statistics-and-plotting.html"><a href="descriptive-statistics-and-plotting.html#the-cancer-gene-atlas-tcga---pancreatic-ductal-adenocarcinoma-data-set"><i class="fa fa-check"></i><b>4.1</b> The Cancer Gene Atlas (TCGA) - Pancreatic Ductal Adenocarcinoma Data Set</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="descriptive-statistics-and-plotting.html"><a href="descriptive-statistics-and-plotting.html#gene-expression-data"><i class="fa fa-check"></i><b>4.1.1</b> Gene expression data</a></li>
<li class="chapter" data-level="4.1.2" data-path="descriptive-statistics-and-plotting.html"><a href="descriptive-statistics-and-plotting.html#clinical-data"><i class="fa fa-check"></i><b>4.1.2</b> Clinical data</a></li>
<li class="chapter" data-level="4.1.3" data-path="descriptive-statistics-and-plotting.html"><a href="descriptive-statistics-and-plotting.html#pdac-subtypes"><i class="fa fa-check"></i><b>4.1.3</b> PDAC Subtypes</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="descriptive-statistics-and-plotting.html"><a href="descriptive-statistics-and-plotting.html#descriptive-plots"><i class="fa fa-check"></i><b>4.2</b> Descriptive plots</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="descriptive-statistics-and-plotting.html"><a href="descriptive-statistics-and-plotting.html#boxplots"><i class="fa fa-check"></i><b>4.2.1</b> Boxplots</a></li>
<li class="chapter" data-level="4.2.2" data-path="descriptive-statistics-and-plotting.html"><a href="descriptive-statistics-and-plotting.html#scatter-plots"><i class="fa fa-check"></i><b>4.2.2</b> Scatter plots</a></li>
<li class="chapter" data-level="4.2.3" data-path="descriptive-statistics-and-plotting.html"><a href="descriptive-statistics-and-plotting.html#hierarchical-clustering"><i class="fa fa-check"></i><b>4.2.3</b> Hierarchical Clustering</a></li>
<li class="chapter" data-level="4.2.4" data-path="descriptive-statistics-and-plotting.html"><a href="descriptive-statistics-and-plotting.html#hierarchical-clustering-using-the-moffitt-transcriptomic-signature"><i class="fa fa-check"></i><b>4.2.4</b> Hierarchical clustering using the Moffitt transcriptomic signature</a></li>
<li class="chapter" data-level="4.2.5" data-path="descriptive-statistics-and-plotting.html"><a href="descriptive-statistics-and-plotting.html#heatmaps"><i class="fa fa-check"></i><b>4.2.5</b> Heatmaps</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="rnaseq-analysis.html"><a href="rnaseq-analysis.html"><i class="fa fa-check"></i><b>5</b> RNAseq Analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="rnaseq-analysis.html"><a href="rnaseq-analysis.html#rnaseq-downstream-analysis"><i class="fa fa-check"></i><b>5.1</b> RNAseq downstream analysis</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="rnaseq-analysis.html"><a href="rnaseq-analysis.html#nf-corernaseq-pipeline---installation"><i class="fa fa-check"></i><b>5.1.1</b> nf-core/rnaseq pipeline - installation</a></li>
<li class="chapter" data-level="5.1.2" data-path="rnaseq-analysis.html"><a href="rnaseq-analysis.html#nextflow"><i class="fa fa-check"></i><b>5.1.2</b> Nextflow</a></li>
<li class="chapter" data-level="5.1.3" data-path="rnaseq-analysis.html"><a href="rnaseq-analysis.html#additional-pipeline-software"><i class="fa fa-check"></i><b>5.1.3</b> Additional pipeline software</a></li>
<li class="chapter" data-level="5.1.4" data-path="rnaseq-analysis.html"><a href="rnaseq-analysis.html#pipeline-code"><i class="fa fa-check"></i><b>5.1.4</b> Pipeline code</a></li>
<li class="chapter" data-level="5.1.5" data-path="rnaseq-analysis.html"><a href="rnaseq-analysis.html#reference-genomes"><i class="fa fa-check"></i><b>5.1.5</b> Reference genomes</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="rnaseq-analysis.html"><a href="rnaseq-analysis.html#running-the-nf-corernaseq-pipeline"><i class="fa fa-check"></i><b>5.2</b> Running the nf-core/rnaseq pipeline</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="rnaseq-analysis.html"><a href="rnaseq-analysis.html#setup"><i class="fa fa-check"></i><b>5.2.1</b> Setup</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="rnaseq-analysis.html"><a href="rnaseq-analysis.html#running-the-nf-corernaseq-pipeline-1"><i class="fa fa-check"></i><b>5.3</b> Running the nf-core/rnaseq pipeline</a></li>
<li class="chapter" data-level="5.4" data-path="rnaseq-analysis.html"><a href="rnaseq-analysis.html#quality-control-qc-and-merged-count-files"><i class="fa fa-check"></i><b>5.4</b> Quality Control (QC) and Merged count files</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="rnaseq-analysis.html"><a href="rnaseq-analysis.html#qc-metrics"><i class="fa fa-check"></i><b>5.4.1</b> QC Metrics</a></li>
<li class="chapter" data-level="5.4.2" data-path="rnaseq-analysis.html"><a href="rnaseq-analysis.html#merged-count-files"><i class="fa fa-check"></i><b>5.4.2</b> Merged count files</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="rnaseq-analysis-deseq2.html"><a href="rnaseq-analysis-deseq2.html"><i class="fa fa-check"></i><b>6</b> RNAseq Analysis - DESeq2</a>
<ul>
<li class="chapter" data-level="6.1" data-path="rnaseq-analysis-deseq2.html"><a href="rnaseq-analysis-deseq2.html#deseq2-package"><i class="fa fa-check"></i><b>6.1</b> DESEq2 package</a></li>
<li class="chapter" data-level="6.2" data-path="rnaseq-analysis-deseq2.html"><a href="rnaseq-analysis-deseq2.html#the-deseqdataset"><i class="fa fa-check"></i><b>6.2</b> The DESeqDataSet</a></li>
<li class="chapter" data-level="6.3" data-path="rnaseq-analysis-deseq2.html"><a href="rnaseq-analysis-deseq2.html#count-matrix-input"><i class="fa fa-check"></i><b>6.3</b> Count matrix input</a></li>
<li class="chapter" data-level="6.4" data-path="rnaseq-analysis-deseq2.html"><a href="rnaseq-analysis-deseq2.html#pre-filtering"><i class="fa fa-check"></i><b>6.4</b> Pre-filtering</a></li>
<li class="chapter" data-level="6.5" data-path="rnaseq-analysis-deseq2.html"><a href="rnaseq-analysis-deseq2.html#differential-expression-analysis"><i class="fa fa-check"></i><b>6.5</b> Differential expression analysis</a></li>
<li class="chapter" data-level="6.6" data-path="rnaseq-analysis-deseq2.html"><a href="rnaseq-analysis-deseq2.html#gene-annotation"><i class="fa fa-check"></i><b>6.6</b> Gene Annotation</a></li>
<li class="chapter" data-level="6.7" data-path="rnaseq-analysis-deseq2.html"><a href="rnaseq-analysis-deseq2.html#exploring-the-results"><i class="fa fa-check"></i><b>6.7</b> Exploring the results</a></li>
<li class="chapter" data-level="6.8" data-path="rnaseq-analysis-deseq2.html"><a href="rnaseq-analysis-deseq2.html#exporting-results"><i class="fa fa-check"></i><b>6.8</b> Exporting results</a></li>
<li class="chapter" data-level="6.9" data-path="rnaseq-analysis-deseq2.html"><a href="rnaseq-analysis-deseq2.html#data-transformation-and-visualisation"><i class="fa fa-check"></i><b>6.9</b> Data transformation and visualisation</a>
<ul>
<li class="chapter" data-level="6.9.1" data-path="rnaseq-analysis-deseq2.html"><a href="rnaseq-analysis-deseq2.html#pca-using-transformed-data"><i class="fa fa-check"></i><b>6.9.1</b> PCA using transformed data</a></li>
<li class="chapter" data-level="6.9.2" data-path="rnaseq-analysis-deseq2.html"><a href="rnaseq-analysis-deseq2.html#heatmaps-using-transformed-data"><i class="fa fa-check"></i><b>6.9.2</b> Heatmaps using transformed data</a></li>
</ul></li>
<li class="chapter" data-level="6.10" data-path="rnaseq-analysis-deseq2.html"><a href="rnaseq-analysis-deseq2.html#gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>6.10</b> Gene Set Enrichment Analysis</a>
<ul>
<li class="chapter" data-level="6.10.1" data-path="rnaseq-analysis-deseq2.html"><a href="rnaseq-analysis-deseq2.html#kegg"><i class="fa fa-check"></i><b>6.10.1</b> KEGG</a></li>
<li class="chapter" data-level="6.10.2" data-path="rnaseq-analysis-deseq2.html"><a href="rnaseq-analysis-deseq2.html#reactome"><i class="fa fa-check"></i><b>6.10.2</b> REACTOME</a></li>
<li class="chapter" data-level="6.10.3" data-path="rnaseq-analysis-deseq2.html"><a href="rnaseq-analysis-deseq2.html#plotting-functions"><i class="fa fa-check"></i><b>6.10.3</b> Plotting functions</a></li>
<li class="chapter" data-level="6.10.4" data-path="rnaseq-analysis-deseq2.html"><a href="rnaseq-analysis-deseq2.html#extracting-enrichment-results-to-generate-heatmaps"><i class="fa fa-check"></i><b>6.10.4</b> Extracting enrichment results to generate heatmaps</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="cancer-dependency-map-depmap.html"><a href="cancer-dependency-map-depmap.html"><i class="fa fa-check"></i><b>7</b> Cancer Dependency Map - DepMap</a>
<ul>
<li class="chapter" data-level="7.1" data-path="cancer-dependency-map-depmap.html"><a href="cancer-dependency-map-depmap.html#downloading-depmap-data-using-the-depmap-package"><i class="fa fa-check"></i><b>7.1</b> Downloading DepMap data using the <code>depmap</code> package</a></li>
<li class="chapter" data-level="7.2" data-path="cancer-dependency-map-depmap.html"><a href="cancer-dependency-map-depmap.html#exploring-depmap-transcriptomic-data"><i class="fa fa-check"></i><b>7.2</b> Exploring DepMap transcriptomic data</a></li>
<li class="chapter" data-level="7.3" data-path="cancer-dependency-map-depmap.html"><a href="cancer-dependency-map-depmap.html#exploring-depmap-mutation-calls"><i class="fa fa-check"></i><b>7.3</b> Exploring DepMap mutation calls</a></li>
<li class="chapter" data-level="7.4" data-path="cancer-dependency-map-depmap.html"><a href="cancer-dependency-map-depmap.html#exploring-depmap-crispr-data"><i class="fa fa-check"></i><b>7.4</b> Exploring DepMap CRISPR data</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">PGT Bioinformatics Course</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="rnaseq-analysis" class="section level1" number="5">
<h1><span class="header-section-number">Chapter 5</span> RNAseq Analysis</h1>
<p>RNAseq (RNA-sequencing) is a technique that can examine the quantity and sequences of RNA in a sample using next generation sequencing (NGS). It analyzes the transcriptome of gene expression patterns encoded within our RNA.</p>
<p>The following course provides an excellent overview of Next Generation Sequencing (NGS) and Illumina NGS Sample Preparation. Please take the time to watch the videos!</p>
<p><strong>Next Generation Sequencing 1 - Overview</strong></p>
<iframe width="750" height="500" src="https://www.youtube.com/embed/mI0Fo9kaWqo" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen>
</iframe>
<p><br></p>
<p><strong>Next Generation Sequencing 2 - Sample Preparation</strong></p>
<iframe width="750" height="500" src="https://www.youtube.com/embed/PFwSe09dJX0" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen>
</iframe>
<p><br></p>
<blockquote>
<p><em>In this talk, Eric Chow explains the chemistry behind next generation sequencing, and describes how the next gen sequencers detect and display results. The most commonly used Illumina sequencers are image based and detect the addition of fluorescently labelled nucleotides. Chow also describes two different next generation sequencing technologies which provide benefits such as much longer reads but with downsides such as higher error rates. Chow finishes the talk with some insights into medical applications of next gen sequencing such as much less invasive prenatal testing or cancer detection.</em></p>
<p><em>In his second talk, Chow discusses Illumina NGS Sample Preparation. He goes over DNA and RNA preparation, bead-based (Ampure or SPRI) cleanups, and sample quantification and quality control.</em></p>
<p>— <a href="https://www.ibiology.org/techniques/next-generation-sequencing/" class="uri">https://www.ibiology.org/techniques/next-generation-sequencing/</a></p>
</blockquote>
<div id="rnaseq-downstream-analysis" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> RNAseq downstream analysis</h2>
<div id="nf-corernaseq-pipeline---installation" class="section level3" number="5.1.1">
<h3><span class="header-section-number">5.1.1</span> nf-core/rnaseq pipeline - installation</h3>
<p><a href="https://nf-co.re/"><img src="images/nf-core-logo.svg" width="300"></a></p>
<p><strong>nf-core/rnaseq</strong> is a bioinformatics analysis pipeline used for RNA sequencing data. The pipeline is built using <strong>Nextflow</strong>, a workflow tool to run tasks across multiple compute infrastructures in a portable manner. Detailed documentation on how to install and run nextflow and nf-core bioinformatic pipelines can be found at the following sites:</p>
<ul>
<li><a href="https://nextflow.io/">https://nextflow.io/</a></li>
<li><a href="https://nf-co.re/">https://nf-co.re/</a></li>
</ul>
<p>For your convenience install instructions are reproduced below.</p>
</div>
<div id="nextflow" class="section level3" number="5.1.2">
<h3><span class="header-section-number">5.1.2</span> Nextflow</h3>
<p><a href="https://nextflow.io/"><img src="images/nextflow2014_no-bg.png" width="300"></a></p>
<p>All nf-core pipelines use Nextflow, so this must be present on the system where you launch your analysis. See <a href="https://nextflow.io/">nextflow.io</a> for the latest installation instructions.</p>
<p>Nextflow runs on most POSIX systems (Linux, Mac OSX etc) and can typically be installed by running the following commands:</p>
<pre><code># Make sure that Java v8+ is installed:
java -version

# Install Nextflow
curl -fsSL get.nextflow.io | bash

# Add Nextflow binary to your user&#39;s PATH:
mv nextflow ~/bin/
# OR system-wide installation:
# sudo mv nextflow /usr/local/bin</code></pre>
<p>You can also install Nextflow using <a href="https://bioconda.github.io/">Bioconda</a>:</p>
<pre><code>conda install -c bioconda nextflow</code></pre>
</div>
<div id="additional-pipeline-software" class="section level3" number="5.1.3">
<h3><span class="header-section-number">5.1.3</span> Additional pipeline software</h3>
<p>nf-core pipelines utilise Docker, Singularity and Conda to seamlessly install and run different software packages. These packages create isolated compute environments on the fly allowing for flexibility in development and delivery.</p>
<ul>
<li><a href="https://docs.docker.com/get-docker/">Docker</a> |
Typically used locally / on single-user servers and the cloud.
Analysis runs in a container, which behaves like an isolated operating system
Previously required system root access, though a “rootless mode” is available since late 2019</li>
<li><a href="https://sylabs.io/docs/">Singularity</a> |
Often used as an alternative to Docker on multi-user systems such as HPC systems.
Also runs containers and can create these from Docker images
Does not need root access or any daemon processes - images built from files</li>
<li><a href="https://docs.conda.io/en/latest/">Conda</a> |
Packaging system that manages environments instead of running analysis in containers.
Poorer reproducibility than Docker / Singularity
There can be changes in low-level package dependencies over time
The software still runs in your native operating system environment and so core system functions can differ</li>
</ul>
</div>
<div id="pipeline-code" class="section level3" number="5.1.4">
<h3><span class="header-section-number">5.1.4</span> Pipeline code</h3>
<p>The nf-core pipeline does not need to be installed. On calling <code>nextflow nf-core/rnaseq ...</code>, pipeline components are automatically downloaded from github.</p>
</div>
<div id="reference-genomes" class="section level3" number="5.1.5">
<h3><span class="header-section-number">5.1.5</span> Reference genomes</h3>
<p>Some pipelines come with built-in support for <a href="https://ewels.github.io/AWS-iGenomes">iGenomes</a> references. It may be preferable for you to download a local copy of these to your system so as to avoid fetching the same reference many times. For more information, see <a href="https://nf-co.re/usage/reference_genomes.md">Reference genomes</a>.</p>
</div>
</div>
<div id="running-the-nf-corernaseq-pipeline" class="section level2" number="5.2">
<h2><span class="header-section-number">5.2</span> Running the nf-core/rnaseq pipeline</h2>
<p>Running the nf-core/rnaseq pipeline is incredibly simple and requires only a basic setup.</p>
<div id="setup" class="section level3" number="5.2.1">
<h3><span class="header-section-number">5.2.1</span> Setup</h3>
<p>The nf-core/rnaseq pipeline typically requires a sample sheet as input. The sample sheet should comprise the following columns.</p>
<table>
<colgroup>
<col width="50%" />
<col width="50%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Column</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">group</td>
<td align="left">Group identifier for sample. This will be identical for replicate samples from the same experimental group.</td>
</tr>
<tr class="even">
<td align="left">replicate</td>
<td align="left">Integer representing replicate number. Must start from 1..<number of replicates>.</td>
</tr>
<tr class="odd">
<td align="left">fastq_1</td>
<td align="left">Full path to FastQ file for read 1. File has to be zipped and have the extension “.fastq.gz” or “.fq.gz.”</td>
</tr>
<tr class="even">
<td align="left">fastq_2</td>
<td align="left">Full path to FastQ file for read 2. File has to be zipped and have the extension “.fastq.gz” or “.fq.gz.”</td>
</tr>
<tr class="odd">
<td align="left">strandedness</td>
<td align="left">Sample strand-specificity. Must be one of unstranded, forward or reverse.</td>
</tr>
</tbody>
</table>
<p><br></p>
<p>A final design file consisting of paired-end data may look something like the one below - see Table 3.1. This sample sheet represents a full sized RNAseq dataset obtained from the ENCODE project, which is routinely used to test the nf-core/rnaseq pipeline and is described <a href="https://github.com/nf-core/test-datasets/tree/rnaseq#full-test-dataset-origin">here</a></p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="rnaseq-analysis.html#cb57-1" aria-hidden="true" tabindex="-1"></a>sampleSheet <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">&quot;RNASeqData/samplesheet.valid.csv&quot;</span>, </span>
<span id="cb57-2"><a href="rnaseq-analysis.html#cb57-2" aria-hidden="true" tabindex="-1"></a>                          <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">&quot;,&quot;</span>)</span>
<span id="cb57-3"><a href="rnaseq-analysis.html#cb57-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sampleSheet)</span>
<span id="cb57-4"><a href="rnaseq-analysis.html#cb57-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;          sample single_end</span></span>
<span id="cb57-5"><a href="rnaseq-analysis.html#cb57-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 GM12878_R1_T1          0</span></span>
<span id="cb57-6"><a href="rnaseq-analysis.html#cb57-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 GM12878_R2_T1          0</span></span>
<span id="cb57-7"><a href="rnaseq-analysis.html#cb57-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3      H1_R1_T1          0</span></span>
<span id="cb57-8"><a href="rnaseq-analysis.html#cb57-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4      H1_R2_T1          0</span></span>
<span id="cb57-9"><a href="rnaseq-analysis.html#cb57-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5    K562_R1_T1          0</span></span>
<span id="cb57-10"><a href="rnaseq-analysis.html#cb57-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6    K562_R2_T1          0</span></span>
<span id="cb57-11"><a href="rnaseq-analysis.html#cb57-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                                                                fastq_1</span></span>
<span id="cb57-12"><a href="rnaseq-analysis.html#cb57-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 s3://nf-core-awsmegatests/rnaseq/input_data/SRX1603629_T1_1.fastq.gz</span></span>
<span id="cb57-13"><a href="rnaseq-analysis.html#cb57-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 s3://nf-core-awsmegatests/rnaseq/input_data/SRX1603630_T1_1.fastq.gz</span></span>
<span id="cb57-14"><a href="rnaseq-analysis.html#cb57-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3 s3://nf-core-awsmegatests/rnaseq/input_data/SRX2370468_T1_1.fastq.gz</span></span>
<span id="cb57-15"><a href="rnaseq-analysis.html#cb57-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4 s3://nf-core-awsmegatests/rnaseq/input_data/SRX2370469_T1_1.fastq.gz</span></span>
<span id="cb57-16"><a href="rnaseq-analysis.html#cb57-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5 s3://nf-core-awsmegatests/rnaseq/input_data/SRX1603392_T1_1.fastq.gz</span></span>
<span id="cb57-17"><a href="rnaseq-analysis.html#cb57-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6 s3://nf-core-awsmegatests/rnaseq/input_data/SRX1603393_T1_1.fastq.gz</span></span>
<span id="cb57-18"><a href="rnaseq-analysis.html#cb57-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                                                                fastq_2</span></span>
<span id="cb57-19"><a href="rnaseq-analysis.html#cb57-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 s3://nf-core-awsmegatests/rnaseq/input_data/SRX1603629_T1_2.fastq.gz</span></span>
<span id="cb57-20"><a href="rnaseq-analysis.html#cb57-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 s3://nf-core-awsmegatests/rnaseq/input_data/SRX1603630_T1_2.fastq.gz</span></span>
<span id="cb57-21"><a href="rnaseq-analysis.html#cb57-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3 s3://nf-core-awsmegatests/rnaseq/input_data/SRX2370468_T1_2.fastq.gz</span></span>
<span id="cb57-22"><a href="rnaseq-analysis.html#cb57-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4 s3://nf-core-awsmegatests/rnaseq/input_data/SRX2370469_T1_2.fastq.gz</span></span>
<span id="cb57-23"><a href="rnaseq-analysis.html#cb57-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5 s3://nf-core-awsmegatests/rnaseq/input_data/SRX1603392_T1_2.fastq.gz</span></span>
<span id="cb57-24"><a href="rnaseq-analysis.html#cb57-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6 s3://nf-core-awsmegatests/rnaseq/input_data/SRX1603393_T1_2.fastq.gz</span></span>
<span id="cb57-25"><a href="rnaseq-analysis.html#cb57-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   strandedness</span></span>
<span id="cb57-26"><a href="rnaseq-analysis.html#cb57-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1      reverse</span></span>
<span id="cb57-27"><a href="rnaseq-analysis.html#cb57-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2      reverse</span></span>
<span id="cb57-28"><a href="rnaseq-analysis.html#cb57-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3      reverse</span></span>
<span id="cb57-29"><a href="rnaseq-analysis.html#cb57-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4      reverse</span></span>
<span id="cb57-30"><a href="rnaseq-analysis.html#cb57-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5      reverse</span></span>
<span id="cb57-31"><a href="rnaseq-analysis.html#cb57-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6      reverse</span></span></code></pre></div>
</div>
</div>
<div id="running-the-nf-corernaseq-pipeline-1" class="section level2" number="5.3">
<h2><span class="header-section-number">5.3</span> Running the nf-core/rnaseq pipeline</h2>
<p>Once the setup is complete, running the nf-core/rnaseq pipeline is as simple as typing the following into the command line:</p>
<pre><code>  nextflow run nf-core/rnaseq \
      --input samplesheet.csv \
      --fasta &#39;&lt;PATH TO FASTA FILE&gt;/genome.fa&#39;\
      --gtf &#39;&lt;PATH TO FASTA FILE&gt;/genome.gtf&#39;
      --aligner star_rsem
      -profile conda</code></pre>
</div>
<div id="quality-control-qc-and-merged-count-files" class="section level2" number="5.4">
<h2><span class="header-section-number">5.4</span> Quality Control (QC) and Merged count files</h2>
<p>After run completion, the nf-core pipeline will generate a number of data files that can be used to assess sequence quality and for downstream analysis. A typical run folder will comprise the following folder structure. Please see <a href="https://nf-co.re/rnaseq/results#rnaseq/results-3643a94411b65f42bce5357c5015603099556ad9/star_rsem/">Run Folder Example.</a></p>
<div id="qc-metrics" class="section level3" number="5.4.1">
<h3><span class="header-section-number">5.4.1</span> QC Metrics</h3>
<p>The first step after the completion of any analysis run is to check the QC metrics. A summary of all QC metrics for a given run can be found in the file named <a href="RNASeqData/multiqc_report.html">multiqc_report.html</a>, which is located in the MultiQC folder. A detailed overview of the content provided in the MultiQC report can be found at the following location <a href="https://nf-co.re/rnaseq/3.0/output">https://nf-co.re/rnaseq/3.0/output</a></p>
<p><a href="./RNASeqData/multiqc_report.html"><img src="images/Multiqc.png"></a></p>
</div>
<div id="merged-count-files" class="section level3" number="5.4.2">
<h3><span class="header-section-number">5.4.2</span> Merged count files</h3>
<p>After assessing QC, we are ready to start a primary analysis of the data. To do so we will typically use a merged gene count file. For example, <a href="RNASeqData/rsem.merged.gene_counts.tsv">rsem.merged.gene_counts.tsv</a>. These files typically comprise the Ensembl Gene ID, Ensembl Transcript ID and Sample IDs with each column/row entry representing the read count for a given gene. See example below:</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="rnaseq-analysis.html#cb59-1" aria-hidden="true" tabindex="-1"></a>mergedCounts <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">&quot;RNASeqData/rsem.merged.gene_counts.tsv&quot;</span>, </span>
<span id="cb59-2"><a href="rnaseq-analysis.html#cb59-2" aria-hidden="true" tabindex="-1"></a>                          <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb59-3"><a href="rnaseq-analysis.html#cb59-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(mergedCounts)</span>
<span id="cb59-4"><a href="rnaseq-analysis.html#cb59-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [1] &quot;gene_id&quot;          &quot;transcript_id.s.&quot; &quot;GM12878_R1&quot;      </span></span>
<span id="cb59-5"><a href="rnaseq-analysis.html#cb59-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [4] &quot;GM12878_R2&quot;       &quot;H1_R1&quot;            &quot;H1_R2&quot;           </span></span>
<span id="cb59-6"><a href="rnaseq-analysis.html#cb59-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [7] &quot;K562_R1&quot;          &quot;K562_R2&quot;          &quot;MCF7_R1&quot;         </span></span>
<span id="cb59-7"><a href="rnaseq-analysis.html#cb59-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [10] &quot;MCF7_R2&quot;</span></span>
<span id="cb59-8"><a href="rnaseq-analysis.html#cb59-8" aria-hidden="true" tabindex="-1"></a>mergedCounts[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span><span class="sc">:</span><span class="dv">7</span>)]</span>
<span id="cb59-9"><a href="rnaseq-analysis.html#cb59-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;           gene_id GM12878_R1 GM12878_R2 H1_R1 H1_R2 K562_R1</span></span>
<span id="cb59-10"><a href="rnaseq-analysis.html#cb59-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 ENSG00000000003          7          0  8299  4390       3</span></span>
<span id="cb59-11"><a href="rnaseq-analysis.html#cb59-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 ENSG00000000005          0          0    69    30       0</span></span>
<span id="cb59-12"><a href="rnaseq-analysis.html#cb59-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3 ENSG00000000419       4061       3582  1885   796    6194</span></span>
<span id="cb59-13"><a href="rnaseq-analysis.html#cb59-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4 ENSG00000000457       1942       1606   592   409     918</span></span>
<span id="cb59-14"><a href="rnaseq-analysis.html#cb59-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5 ENSG00000000460       1862       1612  3594  1748    2813</span></span></code></pre></div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="descriptive-statistics-and-plotting.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="rnaseq-analysis-deseq2.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["PGT_Bioinformatics_course.pdf", "PGT_Bioinformatics_course.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
